### PR TITLE
Refactor gulpfile

### PR DESCRIPTION
**Pull request by kinday**
Привёл немного в порядок. Открыт к предложениям.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,7 +67,6 @@
     "moment": "^2.10.3",
     "npm-run-all": "^1.2.6",
     "numd": "^2.1.0",
-    "object-assign": "^2.0.0",
```
Comment (iamstarkov) [inline package.json:70]: https://travis-ci.org/iamstarkov/jsunderhood/builds/82030104#L443

Comment (kinday) [inline package.json:70]: Забыл про остальные файлы.

Comment (iamstarkov) [inline package.json:70]: what do you think about `R.merge`? merge has no side-effects http://ramdajs.com/0.17/docs/#merge

File: gulpfile.babel.js [hunk 2]
```diff
@@ -1,112 +1,105 @@
...
-  return path.basename(file.relative, path.extname(file.relative));
-};
+import latestInfo from './dump/latest-info.json';
+import { site } from './package.json';
+
+const d = input => moment(new Date(input)).format("DD MMMM YYYY");
+const unix = input => moment(new Date(input)).unix();
 
-var jadeDefaults = {
+const getBasename = ({ relative }) => path.parse(relative).name;
```
Comment (iamstarkov) [inline gulpfile.babel.js:36]: очень нравится

File: gulpfile.babel.js [hunk 3]
```diff
@@ -1,112 +1,105 @@
...
 
-    if (getBasename(file) === 'README') {
-      url = 'about';
-      title = 'О проекте';
-    }
+let articles = [];
 
-    var author = fs.readJsonSync('./dump/' + title + '.json');
+const articleHarvesting = () => {
+  return through.obj((file, enc, cb) => {
```
Comment (iamstarkov) [inline gulpfile.babel.js]: maybe?

```
const articleHarvesting = () =>
  through.obj((file, enc, cb) => {
…
```

File: gulpfile.babel.js [hunk 4]
```diff
@@ -1,112 +1,105 @@
...
     .pipe(articleHarvesting());
 });
 
-gulp.task('index-page', function() {
-  var groupInGrid = function(state, item, i) {
+gulp.task('index-page', () => {
+  const groupInGrid = (state, item, i) => {
     if (i % 3 === 0) { state.push([]); }
     state[state.length - 1].push(item);
     return state;
```
Comment (iamstarkov) [inline gulpfile.babel.js:103]: what do you think about extracting this function to out scope, and then just using it here?

File: gulpfile.babel.js [hunk 5]
```diff
@@ -124,30 +117,26 @@ gulp.task('index-page', function() {
     .pipe(gulp.dest('dist'));
 });
 
+gulp.task('stats-page', () => gulp.src('layouts/stats.jade')
```
Comment (iamstarkov) [inline gulpfile.babel.js]: can i ask to move all gulp.src to a new line in arrow-func cases?

**Comment by iamstarkov** [comment]
I like how you esify and refactor this gulpfile

File: gulpfile.babel.js (hunk 4 above)
Comment (kinday) [inline gulpfile.babel.js:103]: I’d like to pass it to the locals to process the list inside the template. What do you think?

Comment (iamstarkov) [inline gulpfile.babel.js:103]: good idea. to not mess up with presentation layout in the task file. Also maybe you can pass splitEvery http://ramdajs.com/0.17/docs/#splitEvery

Comment (kinday) [inline gulpfile.babel.js:103]: There’s even better way: CSS.

**Pull request merged**